### ISSUE TITLE
Allow updates using administrator credentials from non administrator accounts

### ISSUE THREAD

**Issue by robbiemu** [warpdotdev/Warp#3947]
### Describe the solution you'd like?

```Text
On the mac, update button appears regardless of account privileges, clicking it either directly installs the update (if administrator) or requests elevated privileges (a username and password modal is presented and after success, update installs.
```

### Is your feature request related to a problem? Please describe.

```Text
Every time there is update, I have to download and overwrite, then relaunch the terminal, or else log out, log in as administrador, and then do the update from warp with no issues. Apps can apply these types of changes by requesting administrator access in an update (which I guess must be a separate process and this is why you are not doing it yet). This is really the standard in the mac environment, and warp should do it.
```

### How important is this feature to you?

2

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by Macaldente** [comment]
@dannyneira I really do not understand your reply to @robbiemu because he requested the other way round. It is not possible to install or update Warp by non-administators anyway. 

We are talking about the absolutely normal way to authorize an application update by the admin-id and admin-password if the Warp user is an ordinary non-admin on a Mac. Every single application on a Mac I can install or update when logged in as a non-admin (= standard user account) by typing admin-id and admin-password. 

It is a standard mode of authorization how every single Mac application asks for the authorization when updating any application – except Warp. Screenshot attached for a Brave browser update – in German.

It is hard to understand as Mac user that this auth-prompt is not shown for updating warp but I rather have to download Warp app image for every single Warp update.

<img width="409" alt="admin authorization prompt for update on Mac by standard user" src="https://github.com/warpdotdev/Warp/assets/87149075/f07aaa9a-e87b-4515-9d3a-5d1a4b232eca">

**Comment by dannyneira** [comment]
Understood @Macaldente, it's something we'd need to implement in Warp. It could be improved, but the manual updates are the only workaround, so we appreciate your patience.

**Comment by ToontjeM** [comment]
You are asking for patience since 2021? 

> Note (Nov 2021): We will work on a fix for this in the future so that multiple profiles can use Warp.

**Comment by dannyneira** [comment]
@ToontjeM yes, since 2021 this hasn't been resolved. I'll bring it up with the team as a blocker for many, but there are to be transparent there is no set eta on resolution. As a possible workaround, please checkout the script on [this comment](https://github.com/warpdotdev/Warp/issues/3021#issuecomment-2055497844).

To anyone else facing this issue, please add a 👍 to the original post at the top or comment with your details, and subscribe if you'd like to be notified.

**Comment by bitspittle** [comment]
(Migrating a comment from an issue that was just marked closed as duplicate)

Warp is amazing, but there's friction every week:
* get a notification
* download Warp from the internet
* manually quit the app
* open the dmg
* drag the icon into the Applications folder
* tell Mac OS "yes I want to replace it"
* reopen Warp
* tell MacOS "yes I'm sure I want to open an app that I downloaded from the internet"

Creating a parent admin account and then working in a user account is a good security practice. I would imagine that since your primary user audience is sophisticated, terminal-using devs, that a higher percent of them than average would be set up with such an admin / user split approach.

I don't have any trouble updating any other dev software on Mac, which makes this issue stand out even more to me.

**Comment by aparker-cityworks** [comment]
I'm also experiencing issues with the update process. On my work computer a have a standard user account. The company uses BeyondTrust to elevate privileges when necessary. The update process works for many different apps and I get a prompt from BeyondTrust to elevate to admin privileges during the update process.
This does not work with Warp.
Warp needs to elevate to admin permissions during updates in a standard way that every other app I use manages to do.

**Comment by dpolivaev** [comment]
Same here. I’m on a non-admin account (company policy), and Warp doesn’t let me elevate via admin credentials when I click the “Upgrade” button. Other apps allow this, but Warp just fails. I stopped using Warp because of this and hope it will be fixed eventually.

**Comment by Tcharlton-Modtek** [comment]
I'm shocked this isn't fixed yet. It's not an issue on ANY other application that I know of. While trying out the script someone posted here, I launched iTerm... and it notified me of the update, asked for my credentials, and updated. Easy Peasy. More and more, I seem to be moving back to iTerm for this frankly stupid reason.

**Comment by dpolivaev** [comment]
Yes, it feels terribly wrong.

**Comment by shrugalic** [comment]
I'm a bit surprised that updating doesn't work for you guys yet. I used to have the same issue (https://github.com/warpdotdev/Warp/issues/3021), but updating Warp with the built-in mechanism has been working for me for a while now (months).

My user is still a Standard user, not an Admin. However, my user had been manually added to the sudoers list for years, so it can still use sudo, without being part of the `admin` group (one could argue it's no longer Standard at this point, but the macOS Users preference still shows it as Standard).

Despite this, upgrading Warp didn't use to work for years, but it has been working for months now. I wonder if Warp changed the update mechanism such that being able to use sudo is enough?

**Comment by dpolivaev** [comment]
@shrugalic for using `sudo` I have to use `su admin_user` first.
It does not work for warp updates.
Regards.

**Comment by shrugalic** [comment]
If anyone wants to try enabling sudo without being in the admin group, this is how:
- Edit the shudders by calling `sudo visudo`. This opens the file in `vi`
- Scroll down to the User specification, and add your user to the list where the `root` user and `admin` group are:
```
# root and users in group wheel can run anything on any machine as any user
root		ALL = (ALL) ALL
youruser        ALL = (ALL) ALL
%admin		ALL = (ALL) ALL
```

With this config, updating Warp works for me. However, security-wise it's _almost_ like being in the admin group, so think about if you want that.

**Comment by Tcharlton-Modtek** [comment]
Soooo... I'm not sure when this happened, but the issue seems to be fixed now! I just updated Warp with a single button press!!!

**Comment by bitspittle** [comment]
> Soooo... I'm not sure when this happened, but the issue seems to be fixed now! I just updated Warp with a single button press!!!

This comment got me hopeful, but I just went through the normal manual download flow over here this morning, so if there's been some sort of fix, it doesn't seem universal.

**Comment by aparker-cityworks** [comment]
No, it's not fixed. Still behaving the same as it always has.

**Comment by dannyneira** [comment]
There is no hardset threshold. But we regularly review the top ones and pick and choose based on a variety of factors like engineering lift required to implement, current priorities in the company, how big of an impact will the feature make across out userbase (the +1s is a proxy for this one), etc.

**Comment by dpolivaev** [comment]
@dannyneira you decide not to care, I decide not to use warp. You win, everyone else looses.

**Comment by dpolivaev** [comment]
I just spent about 3 hours of focused “vibe coding” with Cursor and built a starter app that resolves this issue.
It checks the Warp version, performs the update if needed (showing the admin dialog), and then starts Warp.

You can find it here:
👉 https://github.com/dpolivaev/warp-starter

Hope it helps others get started faster.

**Comment by dpolivaev** [comment]
I just spent about 5 hours of focused “vibe coding” with Cursor and built a starter app that resolves this issue.
It checks the Warp version, performs the update if needed (showing the admin dialog), and then starts Warp.
Compiled, released, and even notarized.

You can find it here:
👉 https://github.com/dpolivaev/warp-starter

Hope it helps others update Warp more easily.

**Comment by GueslyPL** [comment]
Hey all ! 

My org - this is a similar experience with all the engineers I know that use it.

Any ETA or progress on this ?

**Comment by aparker-cityworks** [comment]
Please fix this. This has got to be so easy to fix with one AI prompt.

**Comment by aparker-cityworks** [comment]
@oss-maintainers This issue has been open since Dec 2023 and is a standard macOS UX gap — non-admin users can't update Warp because the app bundle isn't writable, and there's no privilege escalation prompt. The codebase already uses `osascript` with admin privileges for CLI installation (`cli_install.rs`), so the same pattern can be applied to autoupdates.

We have a product spec and tech spec ready to go under `specs/GH3947/` — could this issue be labeled `ready-to-spec`?

**Comment by aparker-cityworks** [comment]
@oss-maintainers @captainsafia 
Please consider labeling this `ready-to-spec`
